### ISSUE TITLE
Unbound axis name in filter_pmap + lax.pmean

### ISSUE THREAD

**Issue by jatentaki** [patrick-kidger/equinox#111]
I tried to parallelize my training code using the standard jax idiom of `pmap`ing the entire `update` function and performing `pmean` reduction on loss and gradients inside `update`. Unfortunately, I get an error about the mapped axis name being unbound. The same code works with `filter_vmap`, which I think is supposed to be API-equivalent, so I believe I am experiencing a bug rather than mere misuse. The reproducing code can be found [here](https://gist.github.com/jatentaki/779c949b1ff4e9ea1828fd7098fb0792).
